### ISSUE TITLE
Include `ommers_hash` in `SealedBlock`

### ISSUE THREAD

**Issue by tcoratger** [paradigmxyz/reth#5164]
### Describe the feature

In the following function:

https://github.com/paradigmxyz/reth/blob/0f254e16a50a6d39543fe59590227a09fd0e21bb/crates/consensus/common/src/validation.rs#L196-L208

the `ommers_hash` is calculated inside the function using the `calculate_ommers_root` method. Why not directly integrate directly this `ommers_hash`field into the `SealedBlock` structure. This could be useful in this function on the one hand but also in other future implementations, what do you think?

**Comment by mattsse** [comment]
the ommers hash is now redundant but I'm wondering why the `calculate_ommers_root` does not check for an empty input list

https://github.com/paradigmxyz/reth/blob/0f254e16a50a6d39543fe59590227a09fd0e21bb/crates/primitives/src/proofs.rs#L104-L108

we also have some debt here where the same consts are defined twice

https://github.com/paradigmxyz/reth/blob/0f254e16a50a6d39543fe59590227a09fd0e21bb/crates/primitives/src/proofs.rs#L14-L20

https://github.com/paradigmxyz/reth/blob/0f254e16a50a6d39543fe59590227a09fd0e21bb/crates/primitives/src/constants/mod.rs#L98-L105

**Comment by mattsse** [comment]
> the ommers_hash is calculated inside the function using the calculate_ommers_root method. Why not directly integrate directly this ommers_hashfield into the SealedBlock structure.

it's already part of the header, here we just confirming that the field in the header is actually correct

**Comment by mattsse** [comment]
not sure about the todo, looks outdated/wrong

**Comment by tcoratger** [comment]
@mattsse I see, so I propose to solve that by:
- Removing the outdated `todo` comment
- Checking for empty `[Header]` list in `calculate_ommers_root` in order to return properly the `EMPTY_ROOT` directly without calling the `encode_list` unnecessarily
- Eliminating the constants from the `proof.rs` file to keep only those in the crate `primitives` and standardize everything

What do you think?

**Comment by mattsse** [comment]
> without calling the encode_list unnecessarily

yeah newer blocks will benefit from that because this will prevent encoding an empty list header

> Eliminating the constants from the proof.rs file to keep only those in the crate primitives and standardize everything

sgtm

this comment is also wrong:

https://github.com/paradigmxyz/reth/blob/0f254e16a50a6d39543fe59590227a09fd0e21bb/crates/primitives/src/constants/mod.rs#L103-L105

because this is the hash of an empty __trie__

and also, this function is unused and can be removed

https://github.com/paradigmxyz/reth/blob/0f254e16a50a6d39543fe59590227a09fd0e21bb/crates/primitives/src/proofs.rs#L95-L101

**Comment by DoTheBestToGetTheBest** [comment]
Hey if this issue is free, i can take it :)
